src/jit/nodeinfo.h

```diff
@@ -110,6 +132,22 @@ class TreeNodeInfo
     unsigned char dstCandsIndex;
     unsigned char internalCandsIndex;
 
+#ifdef _TARGET_XARCH_
+private:
+    // Maximum number of source operands that could be in memory.
+    // On xarch at the most one operand could be a memory-op.
+    // That is _srcMemOpCount doesn't exceed 1. In future when
+    // there is a need to exceed this limit, the bit size
+    // width of this member needs to be adjusted accordingly.
+    //
+    // Right now this field is needed only on xarch and in 
+    // future when there is a need to enable reg-optional
+    // operands on other targets, this #ifdef needs to be
+    // adjusted accordingly.
+    unsigned char _srcMemOpCount : 1;
```

Comment:
Maybe we should add a compile-time assert that the size of TreeNodeInfo doesn't increase, to make size increases completely obvious?

---
In reply to: [77708224](https://github.com/dotnet/coreclr/pull/7066#discussion_r77708224) [](ancestors = 77708224)